### PR TITLE
HBASE-28835 Make connector support for Decimal type

### PR DESCRIPTION
**Pull request by duanyyyyyyy**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 38s |  master passed  |
| +1 :green_heart: |  compile  |   0m 42s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 15s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   0m 52s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  scalac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   0m 11s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   0m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 34s |  hbase-spark in the patch passed.  |
|  |   |  16m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/136 |
| Optional Tests | dupname scalac scaladoc unit spotless compile |
| uname | Linux 06ac8b61c8aa 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 166b764 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/1/testReport/ |
| Max. process+thread count | 942 (vs. ulimit of 12500) |
| modules | C: spark/hbase-spark U: spark/hbase-spark |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/1/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Reidddddd** [comment]
Could add some tests to verify this data type could function well?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 30s |  master passed  |
| +1 :green_heart: |  compile  |   0m 43s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 15s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   1m 29s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 12s |  the patch passed  |
| +1 :green_heart: |  scalac  |   1m 12s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   0m 21s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   1m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 53s |  hbase-spark in the patch passed.  |
|  |   |  18m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/2/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/136 |
| Optional Tests | dupname scalac scaladoc unit spotless compile |
| uname | Linux 722ea14f8cc3 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 166b764 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/2/testReport/ |
| Max. process+thread count | 944 (vs. ulimit of 12500) |
| modules | C: spark/hbase-spark U: spark/hbase-spark |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/2/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by duanyyyyyyy** [comment]
> Could add some tests to verify this data type could function well?

done

File: spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala [hunk 1]
```diff
@@ -990,7 +992,8 @@ class DefaultSourceSuite
                      |"DOUBLE_FIELD":{"cf":"c", "col":"double", "type":"double"},
                      |"DATE_FIELD":{"cf":"c", "col":"date", "type":"date"},
                      |"TIMESTAMP_FIELD":{"cf":"c", "col":"timestamp", "type":"timestamp"},
-                     |"STRING_FIELD":{"cf":"c", "col":"string", "type":"string"}
+                     "STRING_FIELD":{"cf":"c", "col":"string", "type":"string"},
```
Comment (NihalJain) [inline spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala:995]: should not this be `|"STRING_FIELD":{"cf":"c", "col":"string", "type":"string"},`

File: spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala [hunk 2]
```diff
@@ -1030,6 +1033,7 @@ class DefaultSourceSuite
     assert(Math.abs(result.get(8).asInstanceOf[Date].getTime - timestamp) <= 86400000)
     assert(result.get(9).asInstanceOf[Timestamp].getTime == timestamp)
     assert(result.get(10) == "string")
+    assert(result.get(11) == decimal)
```
Comment (NihalJain) [inline spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala:1036]: can we also assert on value here?

File: spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala (hunk 1 above)
Comment (duanyyyyyyy) [reply spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala:995]: Sorry for typo

File: spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala (hunk 2 above)
Comment (duanyyyyyyy) [reply spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala:1036]: This is already assert on value....

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 44s |  master passed  |
| +1 :green_heart: |  compile  |   0m 51s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 23s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   1m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  scalac  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   0m 14s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  scaladoc  |   1m  1s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 39s |  hbase-spark in the patch passed.  |
|  |   |  17m 39s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/3/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/136 |
| Optional Tests | dupname scalac scaladoc unit spotless compile |
| uname | Linux 0dffe55b9c8c 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 166b764 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/3/testReport/ |
| Max. process+thread count | 943 (vs. ulimit of 12500) |
| modules | C: spark/hbase-spark U: spark/hbase-spark |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-136/3/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala (hunk 2 above)
Comment (NihalJain) [reply spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala:1036]: Ah my bad, I confused we are assessing data type here. Thanks for the clarification.

**Review by NihalJain — APPROVED** [review]

**Review by stoty — APPROVED** [review]
+1 LGTM

**Pull request merged**